converter/adb_usb/matrix.c

```diff
@@ -342,12 +346,20 @@ void adb_mouse_task(void)
     x = xx * mouseacc;
     y = yy * mouseacc;
 
-    // TODO: Fix HID report descriptor for mouse to support finer resolution
-    // Cap our two bytes per axis to one byte.
-    // Easier with a MIN-function, but since -MAX(-a,-b) = MIN(a,b)...
-    // I.E. MIN(MAX(x,-127),127) = -MAX(-MAX(x, -127), -127) = MIN(-MIN(-x,127),127)
-    mouse_report.x = -MAX(-MAX(x, -127), -127);
-    mouse_report.y = -MAX(-MAX(y, -127), -127);
+    if (scroll_enable) {
+        scroll_state -= y;
+        mouse_report.v = scroll_state / scroll_speed;
+        scroll_state %= scroll_speed;
```

Comment:
Could you explain how `scroll_state` work and what its effect is?

Reply:
`scroll_state` stores the remainder of the division between the y axis and `scroll_speed` so that the rate of change on the v axis can be constant across polling intervals.
`scroll_speed` determines how much change in the y axis will affect the v axis.
So if `scroll_speed` is set to 8 and I move the mouse in the y axis +20, the v axis will change -2 `(int)(-y / scroll_speed)` and the remainder of that division is -4 `-y % scroll_speed`. This remainder is stored in `scroll_state` so that on the next polling interval, the movement captured here is accounted for when the next v axis result is calculated.